refactor: refactoring 'EventDate' to 'Scheduling

sitepark-becker opened the pull request:
- Renamed `EventDate` to `Scheduling`. I just think it's more accurate.
- The `Scheduling` class is not just a DTO anymore as I intended it to be a general scheduling class that can also be used outside of the graphql context. 
- Added several utility methods to the `Scheduling` class
- Added dependency `rlanvin/php-rrule` to handle the RRules. 
- Added a `SchedulingSet` class that offers methods to handle multiple `Scheduling`-objects.
- moved the `Scheduling-` classes from namespace `Atoolo\EventsCalendar\Service\GraphQL` to `Atoolo\EventsCalendar`. (Im not sure though, maybe I should move it to `Atoolo\EventsCalendar\Scheduling` ?)

sitepark-veltrup commented:
> The Scheduling class is not just a DTO anymore as I intended it to be a general scheduling class that can also be used outside of the graphql context.

I don't think it's so good if there is no DTO with which we can transport the data. I also don't think it's so good that scheduling is not immutable.

For example, we return scheduling here. In `ResourceEventDateResolver::getSchedulings`, for example, it would be better if it is a DTO and immutable.

I would imagine it that way:
```php

namespace Atoolo\EventsCalendar\Dto\Scheduling;

class Scheduling 
{
   public function __construct(
      public readony  bool $isFullDay,
      public readony  bool $hasStartTime,
      public readony  bool $hasEndTime,
      public readony  DateTime  $start,
      public readony  DateTime  $end,
      public readony  string  $rRule,
      public readony  bool $isMultiDate,
   )
}
```

Presumably also for `schedulingSet`.
The factory can then also be placed in the `Dto\Scheduling` namespace. If necessary, a `SchedulingBuilder` can also be useful.
I would see all calculations as a service.
`Atoolo\EventsCalendar\Services\Scheduling\OccurrencesGenerator` ...

sitepark-becker commented:
- The `Scheduling` class is now a DTO under `Atoolo\EventsCalendar\Dto\Scheduling`
- Added a new `SchedulingBuilder` class
- `SchedulingSet` is kinda obsolete, so I removed it.
- For the calculations like `getOccurencesOfScheduling` there's now a Service `Atoolo\EventsCalendar\Service\Scheduling\SchedulingManager`

sitepark-becker commented:
Ok, should be ready to merge now.
